### PR TITLE
English Tutorial For LOJ-1041

### PR DESCRIPTION
**Pull request by Evan-CSE**
Thanks @showrav-ansary  for helping me to upload it into github.

### CODE REVIEW

File: 1041/en.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+Key point of this problem is to traverse the whole graph. We can traverse the whole graph using DFS/BFS algorithm. I have implemented this code using BFS. But you can use DFS too. If you are not familiar with any of them then you are requested to learn about BFS/DFS algorithm. 
+
+Lets observe our algorithm. We need to know if all the cities can be traversed at a single traversal. How can we trace if all the cities are covered or not? To do this we can trace each city and put them at SET data structure so that we can be sure about number of different cities. You can do this without help of SET. You are free to do any of the operations to trace them. But I prefer to use SET data structure to trace distinct cities. 
+
+After that we can traverse the graph. Each time we encounter a new node/city we will increase value of counter. After applying graph traversing algorithm our counter will store number of cities/nodes visited so far. Now we can compare this counter with size of SET. If our counter equals to size of the SET then we can be sure that all the cities are connected with each other. But if they are not equal then obviously we should print **impossible**.
```
Comment (rebornplusplus) [inline 1041/en.md:24]: Implementation detail. Seems redundant to me. You can just remove this portion.

Comment (rebornplusplus) [inline 1041/en.md:24]: Implementation detail. Seems redundant to me. You can just remove this portion.

File: 1041/en.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
+## Prerequisite:
```
Comment (rebornplusplus) [inline 1041/en.md:1]: Add the Problem's name and number as title in the first line.

File: 1041/en.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+```
+#include <bits/stdc++.h>
+
+using namespace std;
+
+#define fast                          \
+    ios_base::sync_with_stdio(false); \
+    cin.tie(NULL);                    \
+    cout.tie(NULL);
```
Comment (rebornplusplus) [inline 1041/en.md:39]: Lose the definition, since it's not being used anywhere.

File: 1041/en.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+- [BFS] (https://cp-algorithms.com/graph/breadth-first-search.html)
+- [DFS] (https://cp-algorithms.com/graph/depth-first-search.html) 
+- [Kruskal] (https://cp-algorithms.com/graph/mst_kruskal.html) 
+- [Prim's] (https://cp-algorithms.com/graph/mst_prim.html)
+
+
+### Tutorial by:
+Md Hasibur Rahman (Evan) <br/>
+Department of CSE, <br/>
+Jatiya Kabi Kazi Nazrul Islam University
```
Comment (rebornplusplus) [inline 1041/en.md:160]: Add your LightOJ Profile link here instead, and keep these information in your profile page. Would be better organized.

**Review by rebornplusplus — CHANGES_REQUESTED** [review]
Hi @Evan-CSE, good efforts. I have a few comments about some of the stuffs, see to those. Then we can merge the PR.

File: 1041/en.md (hunk 2 above)
Comment (Evan-CSE) [reply 1041/en.md:1]: Done

File: 1041/en.md (hunk 3 above)
Comment (Evan-CSE) [reply 1041/en.md:39]: Done

File: 1041/en.md (hunk 4 above)
Comment (Evan-CSE) [reply 1041/en.md:160]: Done

**Comment by Evan-CSE** [comment]
Dear @rebornplusplus , can you recheck it?

File: 1041/en.md [hunk 5]
```diff
@@ -1,3 +1,6 @@
+## Problem Id: LOJ-1041
+## Problem Name: Road Construction
```
Comment (rebornplusplus) [inline 1041/en.md:2]: Merge these two into one line, as "LOJ 1041: Road Construction".
Use Header 1, with a single preceding #.

File: 1041/en.md [hunk 6]
```diff
@@ -16,13 +19,6 @@ To solve this problem we need to know if all the cities are directly or indirect
 
 For my convenience I will implement this in a different way.
```
Comment (rebornplusplus) [inline 1041/en.md:20]: Remove this line as it became irrelevant due to prior removal.

**Review by rebornplusplus — CHANGES_REQUESTED** [review]

File: 1041/en.md (hunk 5 above)
Comment (Evan-CSE) [reply 1041/en.md:2]: solved

File: 1041/en.md (hunk 6 above)
Comment (Evan-CSE) [reply 1041/en.md]: solved

**Review by rebornplusplus — APPROVED** [review]
Looks good. Thanks.

**Pull request merged**